### ISSUE TITLE
cli / config switch to turn off video upload

### ISSUE THREAD

**Issue by kjellski** [cypress-io/cypress#2652]
### Current behavior:

There is no way to record a video and not upload them at all.

### Desired behavior:

A `cypress.json` or `cypress` cli switch that enables a user of suppressing the upload while keeping everything else. Like `"videoUpload": false,` or `--video-upload false`.

### Versions

Every version.

**Comment by jennifer-shehane** [comment]
Hey @kjellski, why is it that you want to record the video, but not upload. Just trying to understand the motivation behind your wanting this.

**Comment by kjellski** [comment]
I've created #2653 in order to help resolve this.

**Comment by kjellski** [comment]
After thinking about this a bit more, it might be better to modify the `--record` switch to not require the `dashboard.cypress.io`. Although, I think it should be backwards compatible and change the bails form the missing `key` part to warnings, so that someone would actually know why things are not working for your business usecase @jennifer-shehane . What do you think?

**Comment by brian-mann** [comment]
@kjellski We appreciate the PR and have discussed this internally - but we're confused as to **why** you'd like this option to exist.

For the record, the video plays an important role for recording to the Dashboard - and we are currently working on ways that make it more valuable - such as helping you see the last time a test passed or failed historically - by linking to previous runs. In those situations, the video being uploaded even on passing builds will be helpful. Although we do have the option `videoUploadOnPasses` we may actually end up removing that down the road.

We don't intend to make the `videoUpload` configurable because we're designing features expecting the video to exist. We understand that the current feature set may be limiting - but there are many things that the client sends to the Dashboard that are not exposed via configuration for the same purposes and reasons listed above.

The time it takes the video to upload is fractions of a second, so I can't see why it would be helpful to turn off there either.

Do you mind going into more detail as to the use case / purpose of this?

**Comment by jennifer-shehane** [comment]
Another relevant thread on record versus upload for videos: https://github.com/cypress-io/cypress/issues/2522#issuecomment-424358181

**Comment by kjellski** [comment]
@brian-mann Thanks for your answer! I do understand that this should be the default behaviour and that it is needed to be the default for you to build features upon.

But what if someone would not be using the dashboard from cypress.io? All usecases are perfectly valid for the case where someone wants to just "drop in" testing with cypress and not set anything up. But since we are for example not using the dashboard at all, we don't need that.

We've only created the account on cypress.io because it's needed in order for videos to be recorded. That is exactly what we want to spare out of our process. 
Basically @GrayedFox said in #2522, he was just better on expressing the initial thought ;) /cc @jennifer-shehane 

I even like his idea more and would aim to implement his switch because it would feel cleaner from the cli perspective.

**Comment by kjellski** [comment]
@brian-mann @jennifer-shehane any feedback on this? I won't implement this if you'd decline the PR anyways, but I think it would be a valuable addition to cypress.

**Comment by brian-mann** [comment]
@kjellski 	I'm coming back to this now, and the answer may be in front of me - but I'm confused still as to why this is necessary. As of today, you don't need to record to the dashboard to get value out of the video. The option `videoUploadOnPasses` only affects when `--record`. If you're not using the Dashboard then you can manage the video themselves since they'll still be physically on disk.

I must be missing something. Your comment here: https://github.com/cypress-io/cypress/issues/2652#issuecomment-433296129 suggests that you're not using the Dashboard, and I just want to make sure we're on the same page that you can completely manage the video yourself if that's the case without us needing to change anything on our end.

**Comment by kjellski** [comment]
Thanks for getting involved again @brian-mann! So the problem in general is this:
AFAIK you can not use video recording without sending it out to your servers via upload. 

So you would be able to only record videos, without upload? I mean that's what I'm after... but it's not possible to use `--record` without a key to the dashboard and then it uploads... maybe I'm missing some documentation that suggested this... :D

**Comment by jennifer-shehane** [comment]
You do not need the `--record` key to have video record. There is output at the end of each spec file when run using `cypress run` that displays where the video was saved locally. This video was not uploaded to Cypress services in any way.

For example, I ran `cypress run`  in my cypress-example-kitchensink project and the output prints:

```

  (Screenshots)

  - /Users/jennifer/Dev/cypress-example-kitchensink/cypress/screenshots/examples/actions.spec.js/Actions -- type() - type into a DOM element -- before each hook (failed).png (1280x720)


  (Video)

  - Started processing:   Compressing to 32 CRF
  - Finished processing:  /Users/jennifer/Dev/cypress-example-kitchensink/cypress/videos/examples/actions.spec.js.mp4 (0 seconds)
```

And in my `cypress/videos` folder I see:

<img width="299" alt="screen shot 2018-12-03 at 3 01 02 pm" src="https://user-images.githubusercontent.com/1271364/49362127-4f7c1400-f70c-11e8-8a83-e1a0c2f2d487.png">

**Comment by kjellski** [comment]
Oh well, if that's the case, that is really nice! I think we might consider improving the documentation with that regard then... 

We've always thought that the --record flag is in place to enable recording. That is why I was asking for the flag in order to turn off the upload.

**Comment by jennifer-shehane** [comment]
It is quite a confusing argument name - I agree. 😞
